backend/MoleculeOSSite/Models/Validators/RegisterValidator.cs

```diff
@@ -0,0 +1,21 @@
+﻿using FluentValidation;
+using MoleculeOSSite.Entities;
+
+namespace MoleculeOSSite.Models.Validators
+{
+    public class RegisterValidator:AbstractValidator<User>
+    {
+        public RegisterValidator(MyDbContext dbcontext)
+        {
+            RuleFor(x=>x.Email).NotEmpty().EmailAddress();
+            RuleFor(x => x.PasswordHash).NotEmpty().MinimumLength(6);
+            RuleFor(x=>x.ConfirmPassword).Equal(u=>u.PasswordHash);
+            RuleFor(x => x.Email).Custom((value, context) =>
+            {
+                var emailExist = dbcontext.Users.Any(u=>u.Email == value);
+                if (emailExist)
+                    context.AddFailure("Email", "Email is already taken");
```

Comment:
```suggestion
    public class RegisterValidator : AbstractValidator<RegisterDTO>
    {
        public RegisterValidator(MyDbContext dbcontext)
        {
            RuleFor(x => x.Email).NotEmpty().EmailAddress();
            RuleFor(x => x.PasswordHash).NotEmpty().MinimumLength(6);
            RuleFor(x => x.ConfirmPassword).Equal(x => x.Password);
            RuleFor(x => x.Email).Custom((value, context) =>
            {
                var emailExist = dbcontext.Users.Any(u => u.Email == value);
                if (emailExist)
                    context.AddFailure(nameof(User.Email), "Email is already taken");
```
1. Nie używaj User jako klasy, która reprezentuje encje w tabeli i jednocześnie służy jako dto z ważnymi danymi do rejestrowania, bo potem wynika z tego wiele problemów i utrudnień. Zamiast tego utwórz jeszcze klasę o nazwie np: RegisterDTO
2. Odstępy
3. Użyj nameof(User.Email) zamiast "Email".